collagraph/component.py

```diff
@@ -0,0 +1,41 @@
+from abc import ABCMeta, abstractmethod
+
+from observ import reactive, readonly
+
+
+class Component(metaclass=ABCMeta):
+    """Abstract base class for components"""
+
+    def __init__(self, props=None):
+        self.props = readonly(props or {})
```

Comment:
An empty dict also evaluates to False, not sure if that might lead to surprising effects

Reply:
Ah, right: in order to preserve reactivity... I'll just check for `is None` instead.